ironic-deployment/base/ironic.yaml

```diff
@@ -169,6 +175,9 @@ spec:
       volumes:
       - name: ironic-data-volume
         emptyDir: {}
+      - name: ironic-credentials
+        secret:
+          secretName: ironic-credentials
```

Comment:
Why do we need this in the `base`? (I'm open to make basic-auth mandatory and part of the base, but then I think we should get rid of the basic-auth component.)
I'm afraid this will make it impossible to deploy without credentials, which can be convenient in development.

Reply:
In trying to resolve the 'metal3-bmo-e2e-test-optional-pull' optional job, I edited this file and forgot to change it back. I have removed the credentials from it now.